test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -909,6 +926,226 @@ TEST_P(ExtProcIntegrationTest, GetBufferedButNoBodies) {
   verifyDownstreamResponse(*response, 200);
 }
 
+TEST_P(ExtProcIntegrationTest, RemoveRequestContentLengthInStreamedMode) {
+  proto_config_.mutable_processing_mode()->set_request_body_mode(ProcessingMode::STREAMED);
+  proto_config_.mutable_processing_mode()->set_response_header_mode(ProcessingMode::SKIP);
+
+  ConfigOptions config_option = {};
+  config_option.http1_codec = true;
+  initializeConfig(config_option);
+  HttpIntegrationTest::initialize();
+
+  auto response =
+      sendDownstreamRequestWithBody("test!", absl::nullopt, /*add_content_length=*/true);
+  processRequestHeadersMessage(*grpc_upstreams_[0], true, absl::nullopt);
+  processRequestBodyMessage(
+      *grpc_upstreams_[0], false, [](const HttpBody& body, BodyResponse& body_resp) {
+        EXPECT_TRUE(body.end_of_stream());
+        auto* body_mut = body_resp.mutable_response()->mutable_body_mutation();
+        body_mut->set_body("Hello, World!");
+        return true;
+      });
+
+  handleUpstreamRequest();
+  EXPECT_EQ(upstream_request_->headers().ContentLength(), nullptr);
+  EXPECT_THAT(upstream_request_->headers(),
+              HeaderValueOf(Http::Headers::get().TransferEncoding,
+                            Http::Headers::get().TransferEncodingValues.Chunked));
+  EXPECT_EQ(upstream_request_->body().toString(), "Hello, World!");
+  verifyDownstreamResponse(*response, 200);
+}
+
+// Test the request content length is removed in BUFFERED BodySendMode + SKIP HeaderSendMode..
+TEST_P(ExtProcIntegrationTest, RemoveRequestContentLengthInBufferedMode) {
+  proto_config_.mutable_processing_mode()->set_request_header_mode(ProcessingMode::SKIP);
+  proto_config_.mutable_processing_mode()->set_request_body_mode(ProcessingMode::BUFFERED);
+  proto_config_.mutable_processing_mode()->set_response_header_mode(ProcessingMode::SKIP);
+
+  initializeConfig();
+  HttpIntegrationTest::initialize();
+
+  auto response =
```

Comment:
Thanks for adding the tests to verify the removing the content-length logic is working in all three cases.

1) It looks to me the common code, like mutate the body, check the upstream request,  can be put into a function and shared by all these tests.
2) Can you add a test for the response path as well?

Reply:
I thought about this. 
I think all recently added tests are added as this way. So, there should be a separate effort to refactor all ext_proc tests rather than here.

Regarding response path, basic functionality is tested in filter/integration test. Given the logic is exactly same as request, I don't think it provide too much value